test/integration_serial_test.go

```diff
@@ -352,3 +352,58 @@ func TestTerragruntParallelism(t *testing.T) {
 		})
 	}
 }
+
+// NOTE: The following test requires at least two service accounts.
+// 1. For running integration tests. This requires storage administrator privileges.
+//  - This is the test runner account set in the environment variable "GOOGLE_APPLICATION_CREDENTIALS".
+//  - This is set to "impersonate_service_account" and used for privilege impersonation.
+//    Set the service account email address in the environment variable "GOOGLE_IDENTITY_EMAIL".
+// 2. For running impersonation tests. This requires token creation privileges without storage admin privileges.
+//  - This is the pre-impersonation account set in the environment variable "GCLOUD_SERVICE_KEY_IMPERSONATOR".
+func TestTerragruntWorksWithImpersonateGCSBackend(t *testing.T) {
+	defaultCreds := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+	defer os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", defaultCreds)
+	// change to impersonator credentials
+	impersonatorKey := os.Getenv("GCLOUD_SERVICE_KEY_IMPERSONATOR")
+	if impersonatorKey == "" {
+		t.Fatalf("Required environment variable `%s` - not found", "GCLOUD_SERVICE_KEY_IMPERSONATOR")
+	}
+	tmpImpersonatorCreds := createTmpTerragruntConfigContent(t, impersonatorKey, "impersonator-key.json")
+	defer removeFile(t, tmpImpersonatorCreds)
+	os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", tmpImpersonatorCreds)
+
+	cleanupTerraformFolder(t, TEST_FIXTURE_GCS_PATH)
+	cleanupTerraformFolder(t, TEST_FIXTURE_GCS_IMPERSONATE_PATH)
+
+	// We need a project to create the bucket in, so we pull one from the recommended environment variable.
+	project := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	gcsBucketName := fmt.Sprintf("terragrunt-test-bucket-%s", strings.ToLower(uniqueId()))
+
+	defer deleteGCSBucket(t, gcsBucketName)
+
+	// run without impersonation
+	tmpTerragruntGCSConfigPath := createTmpTerragruntGCSConfig(t, TEST_FIXTURE_GCS_PATH, project, TERRAFORM_REMOTE_STATE_GCP_REGION, gcsBucketName, config.DefaultTerragruntConfigPath)
+	// run terregrunt ignore errors
+	runTerragruntCommand(t, fmt.Sprintf("terragrunt apply -auto-approve --terragrunt-non-interactive --terragrunt-config %s --terragrunt-working-dir %s", tmpTerragruntGCSConfigPath, TEST_FIXTURE_GCS_PATH), os.Stdout, os.Stderr)
```

Comment:
why error should be ignored? why can't be used `assert.Error()`?

Reply:
Since the command is executed without impersonation, the command will fail due to lack of privileges.
This is expected behavior and the error should be ignored.

Lack of privileges should prevent the bucket from being created, so checking for that here.
https://github.com/gruntwork-io/terragrunt/pull/2550/files/7956720c2c9cc4a17fb6d8de61a54494e4962655#diff-288bea63faddba4478b115bc89ce4fc23b9fab5f9a17dabce38fbb69033a0f7aR393

Reply:
If is expected to have an error, then should be checked that the error is returned, it will help in the future if for some reason behavior of code will change and error will not be thrown, it will also be more to find that error is not returned instead if debugging why GCS bucket wasn't created.

Reply:
Thanks for explaining.
I have modified it to use `assert.Error()`.